papermill/preprocess.py

```diff
@@ -202,3 +194,81 @@ def papermill_preprocess(self, nb, resources):
+
+        while True:
+            try:
+                # We are not waiting for execute_reply, so all output
+                # will not be waiting for us. This may produce currently unknown issues.
+                msg = self.kc.iopub_channel.get_msg(timeout=None)
+            except Empty:
+                self.log.warning("Timeout waiting for IOPub output")
+                if self.raise_on_iopub_timeout:
+                    raise RuntimeError("Timeout waiting for IOPub output")
+                else:
+                    break
+            if msg['parent_header'].get('msg_id') != msg_id:
+                # not an output from our execution
+                continue
+
+            msg_type = msg['msg_type']
+            self.log.debug("output: %s", msg_type)
+            content = msg['content']
+
+            # set the prompt number for the input and the output
+            if 'execution_count' in content:
+                cell['execution_count'] = content['execution_count']
+
+            if msg_type == 'status':
+                if content['execution_state'] == 'idle':
+                    break
+                else:
+                    continue
+            elif msg_type == 'execute_input':
+                sys.stdout.write('Executing Cell {:-<40}\n'.format(content.get("execution_count", "*")))
+                continue
+            elif msg_type == 'clear_output':
+                outs[:] = []
+                # clear display_id mapping for this cell
+                for display_id, cell_map in self._display_id_map.items():
+                    if cell_index in cell_map:
+                        cell_map[cell_index] = []
+                continue
+            elif msg_type.startswith('comm'):
+                continue
+
+            display_id = None
+            if msg_type in {'execute_result', 'display_data', 'update_display_data'}:
+                display_id = msg['content'].get('transient', {}).get('display_id', None)
+                if display_id:
+                    self._update_display_id(display_id, msg)
+                if msg_type == 'update_display_data':
+                    # update_display_data doesn't get recorded
+                    continue
+
+            try:
+                out = output_from_msg(msg)
+            except ValueError:
+                self.log.error("unhandled iopub msg: " + msg_type)
+                continue
+            if display_id:
+                # record output index in:
+                #   _display_id_map[display_id][cell_idx]
+                cell_map = self._display_id_map.setdefault(display_id, {})
+                output_idx_list = cell_map.setdefault(cell_index, [])
+                output_idx_list.append(len(outs))
+
+            if self.log_output:
+                log_output(out)
+            outs.append(out)
+
+        exec_reply = self._wait_for_reply(msg_id, cell)
+        if self.log_output: 
+            sys.stdout.write('Ending Cell {:-<43}\n'.format(
```

Comment:
This should use stderr instead of stdout as it's a message instead of data (for good unix-like command support)